pythonforandroid/toolchain.py

```diff
@@ -587,6 +594,9 @@ def add_parser(subparsers, *args, **kwargs):
 
         self.ctx = Context()
         self.ctx.use_setup_py = getattr(args, "use_setup_py", True)
+        self.ctx.build_as_debuggable = getattr(
+            args, "build_mode", "debug"
```

Comment:
Minor: seems like we're repeating ourselves in a similar fashion few times, enough to make an helper function maybe.
Maybe something like:
```python
def build_as_debuggable(args):
    return getattr(args, "build_mode", "debug") == "release"
```

Reply:
I just tried this, and I have the feeling the actual problem is we want something like #1812 instead so we can do simple `(args.build_mode == "debug")` everywhere and it'll look ok. The problem is that sometimes "release" and "debug" are checked separately and it looks like someone may want to add a mode in these places so dropping this in everywhere would kinda look weird I think, but in only half the places it also looks weird. I think the real problem really is `hasattr`/`getattr` making this look ugly

Reply:
Yes `hasattr`/`getattr` definitely makes it look hugly, but testing for debug or release for me is just like testing for True or False. It's a two state, so you can always negate it. Yes it can also be three state, but I don't see it as a problem.
Anyway the PR looks fine overall and it's unit tested, so I'm also happy to keep it this way